### PR TITLE
fix: sanitizeMarkdown on packageName & prTitle

### PR DESCRIPTION
**Pull request by ThibautMarechal**
## Changes:

Use sanitizeMarkdown for PrTitle & PackageName in the update table & onboarding pr-list.

## Context:

I have a weird behavior using the platform bitbucket-server. Adding the zero-width character after the '@' fixed the issue.
image
it display '@testing/react-autosuggest' but the real packageName is  '@type/react-autosuggest'. 
(@ testing is coming from the package above)
It only hapend when the packageName is in a link.
It's now fixed by the sanitizeMarkdown function.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Please create an issue to discuss the problem and possible solutions first. The solution you propose is now resulting in some double escaping, which is not ideal.

**Comment by JamieMagee** [comment]
@ThibautMarechal Can you rebase your branch onto latest `main` and mark it ready for review? Thanks!

File: lib/workers/pr/body/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { platform } from '../../../../test/util';
+import { getConfig } from '../../../config/defaults';
+import type { RenovateConfig } from '../../../config/types';
+
+import { getPrBody } from '.';
+
+describe('workers/pr/body', () => {
```
Comment (JamieMagee) [inline lib/workers/pr/body/index.spec.ts:7]:
```suggestion
describe(getName(), () => {
```

File: lib/workers/pr/body/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { platform } from '../../../../test/util';
```
Comment (JamieMagee) [inline lib/workers/pr/body/index.spec.ts:1]:
```suggestion
import { getName, platform } from '../../../../test/util';
```

File: lib/workers/pr/body/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { getConfig } from '../../../config/defaults';
+import type { RenovateConfig } from '../../../config/types';
+
+import { getPrBody } from '.';
+
+describe('workers/pr/body', () => {
+  let config: RenovateConfig;
+  beforeEach(() => {
```
Comment (JamieMagee) [inline lib/workers/pr/body/index.spec.ts:9]: You'll probably want to mock `getPrConfigDescription` and `getControls` as these make `git` commands.

Comment (ThibautMarechal) [reply lib/workers/pr/body/index.spec.ts:15]: Never did that before, can you confirm that I did the right things ?

File: lib/workers/pr/body/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import { getConfig } from '../../../config/defaults';
+import type { RenovateConfig } from '../../../config/types';
+import { getPrBody } from '.';
+
+jest.mock('./config-description', () => ({
+  getPrConfigDescription: jest.fn(() => ''),
+}));
+jest.mock('./controls', () => ({
+  getControls: jest.fn(() => ''),
+}));
```
Comment (viceice) [inline lib/workers/pr/body/index.spec.ts:11]:
```suggestion
jest.mock('./config-description');
jest.mock('./controls');
```

just use default mocks jests will do it's magic 🧙

File: lib/workers/pr/body/__snapshots__/index.spec.ts.snap [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`workers/pr/body generate the message markdown 1`] = `
+"This PR contains the following updates:
+
+| Package | Type | Update | Change |
+|---|---|---|---|
+| [@&#8203;scope/package-with-link](http://scope.com/package-with-link) | dependencies | patch | \`^10.1.1\` -> \`^10.1.2\` |
+| @&#8203;scope/package-without-link | dependencies | patch | \`^10.1.1\` -> \`^10.1.2\` |
```
Comment (viceice) [inline lib/workers/pr/body/__snapshots__/index.spec.ts.snap:9]: should we quote this as `@&#8203;scope/package-without-link`

**Comment by JamieMagee** [comment]
Snapshots need to be updated `yarn jest -u`

File: lib/workers/pr/body/__snapshots__/index.spec.ts.snap (hunk 5 above)
Comment (rarkins) [reply lib/workers/pr/body/__snapshots__/index.spec.ts.snap:9]: Maybe in a separate PR

**Review by rarkins — APPROVED** [review]

**Review by rarkins — COMMENTED** [review]
Needs running on a real repo to sanity check it

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs running on a real repo to sanity check it

**Comment by rarkins** [comment]
> Needs running on a real repo to sanity check it

This still applies

**Comment by rarkins** [comment]
@ThibautMarechal were you able to verify this on one or more public repos?

**Pull request closed without merging**